notebooks/work-in-progress/VCERARE.ipynb

```diff
@@ -0,0 +1,288 @@
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# random county_id generator:\n",
+    "random_county = random.choice(rare_df[\"county_id_fips\"])"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "cb3e5f29-7c98-44a2-8578-b64a6878733d",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Editable fields: \n",
+    "year = 2023\n",
+    "county_id = random_county\n",
+    "gen_type = \"onshore_wind\" # choose from: solar_pv, onshore_wind, offshore_wind"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "3915c2de-215f-4b25-b718-15f14ed3dbd6",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Non-editable variables\n",
+    "plot_df = rare_df[\n",
+    "    (rare_df[\"report_year\"]==year)\n",
+    "    & (rare_df[\"county_id_fips\"]==county_id)\n",
+    "]\n",
+    "county_name = plot_df[plot_df[\"county_id_fips\"]==county_id].county_or_lake_name.unique().item()\n",
+    "state_name = plot_df[plot_df[\"county_id_fips\"]==county_id].state.unique().item()\n",
+    "\n",
+    "# Make the chart\n",
+    "plt.hist(plot_df[f\"capacity_factor_{gen_type}\"], bins=100, range=(0, 1))\n",
+    "plt.title(\n",
+    "    f\"county: {county_name}; state: {state_name}; gen: {gen_type}; max:{plot_df[f\"capacity_factor_{gen_type}\"].max():.2f}\")\n",
+    "plt.grid()\n",
+    "plt.show();"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "09d939e8-3a3d-4908-9ca3-67e4c69c35d8",
+   "metadata": {},
+   "source": [
+    "### Download the data"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "13289ced-0028-4d79-9e5a-cf79bc464d80",
+   "metadata": {},
+   "source": [
+    "#### Memory check - will it Excel? \n",
+    "\n",
+    "If you want to know whether your table is capable of being processed as a csv you can use this memory estimator. If the estimated memory exceeds 500 MB it's too big! Cutting columns or making the filter scope smaller will help reduce the file size."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "47400458-83f7-4a6b-abea-55ae9618bf27",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# test_df = the name of the table you want to test\n",
+    "test_df = plot_df\n",
+    "\n",
+    "# Calculate the memory usage in bytes, including the index\n",
+    "mem_usage_bytes = test_df.memory_usage(deep=True).sum()\n",
+    "# Convert bytes to megabytes\n",
+    "mem_usage_mb = mem_usage_bytes / (1024 * 1024)\n",
+    "csv_size_estimate_mb = mem_usage_mb * 2  # Rough multiplier for CSV size\n",
+    "print(f\"Estimated CSV size: {csv_size_estimate_mb:.2f} MB\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "17bff202-4766-44e1-8442-86108335162b",
+   "metadata": {},
+   "source": [
+    "#### Download!\n",
+    "Specify your desired download location by filling in the `download_path` and running this cell will output the data to that location under the name `rare_power_data.csv`."
```

Comment:
We'll need to give people more guidance here, since path names often trip people up. Also, having the slash on the file name will cause problems if people just leave this blank.

I'd suggest:

```
# Add the file path you want to download the data to
# Leave this blank to save the data in the same folder as this notebook.
download_path = "" 
# e.g. download_path = "/home/user/Desktop/folder/data_folder/"
plot_df.to_csv(download_path+"rare_power_data.csv")
```